apps/prairielearn/src/lib/score-upload.js

```diff
@@ -1,316 +1,249 @@
-          let outputThreshold = 100;
+    serverJob.executeInBackground(async (job) => {
+      job.info('Uploading question scores for ' + assessment_label);
 
-          job.verbose(
-            `Parsing uploaded CSV file "${csvFile.originalname}" (${csvFile.size} bytes)`
-          );
-          job.verbose(`----------------------------------------`);
-          const csvStream = streamifier.createReadStream(csvFile.buffer, {
-            encoding: 'utf8',
-          });
-          const csvConverter = csvtojson({
-            colParser: {
-              instance: 'number',
-              score_perc: 'number',
-              points: 'number',
-              manual_score_perc: 'number',
-              manual_points: 'number',
-              auto_score_perc: 'number',
-              auto_points: 'number',
-              submission_id: 'number',
-            },
-            maxRowLength: 10000,
-          });
-          let successCount = 0,
-            errorCount = 0;
-          csvConverter
-            .fromStream(csvStream)
-            .subscribe((json, number) => {
-              return new Promise((resolve, _reject) => {
-                // Replace all keys with their lower-case values
-                json = _.mapKeys(json, (v, k) => {
-                  return k.toLowerCase();
-                });
-                const msg = `Processing CSV record ${number}: ${JSON.stringify(json)}`;
-                if (output == null) {
-                  output = msg;
+      // accumulate output lines in the "output" variable and actually
+      // output put them in blocks, to avoid spamming the updates
+      let output = null;
+      let outputCount = 0;
+      let outputThreshold = 100;
+
+      let successCount = 0;
+      let errorCount = 0;
+
+      job.info(`Parsing uploaded CSV file "${csvFile.originalname}" (${csvFile.size} bytes)`);
+      const csvStream = streamifier.createReadStream(csvFile.buffer, {
+        encoding: 'utf8',
+      });
+      const csvConverter = csvtojson({
+        colParser: {
+          instance: 'number',
+          score_perc: 'number',
+          points: 'number',
+          manual_score_perc: 'number',
+          manual_points: 'number',
+          auto_score_perc: 'number',
+          auto_points: 'number',
+          submission_id: 'number',
+        },
+        maxRowLength: 10000,
+      });
+
+      try {
+        await csvConverter.fromStream(csvStream).subscribe((json, number) => {
+          return new Promise((resolve, _reject) => {
+            // Replace all keys with their lower-case values
+            json = _.mapKeys(json, (v, k) => {
+              return k.toLowerCase();
+            });
+            const msg = `Processing CSV record ${number}: ${JSON.stringify(json)}`;
+            if (output == null) {
+              output = msg;
+            } else {
+              output += '\n' + msg;
+            }
+            module.exports._updateInstanceQuestionFromJson(
```

Comment:
I opted to not convert all this to async/await just yet, I'm only focusing on server jobs in this PR.

Reply:
This is actually done in #6271...